cmd/kubeadm/app/cmd/certs.go

```diff
@@ -332,27 +342,24 @@ func renewCert(kdir string, internalcfg *kubeadmapi.InitConfiguration, handler *
 	return nil
 }
 
-func getInternalCfg(cfgPath string, kubeconfigPath string, cfg kubeadmapiv1.ClusterConfiguration, out io.Writer, logPrefix string) (*kubeadmapi.InitConfiguration, error) {
+func getInternalCfg(cfgPath string, client kubernetes.Interface, cfg kubeadmapiv1.ClusterConfiguration, printer output.Printer, logPrefix string) (*kubeadmapi.InitConfiguration, error) {
 	// In case the user is not providing a custom config, try to get current config from the cluster.
 	// NB. this operation should not block, because we want to allow certificate renewal also in case of not-working clusters
-	if cfgPath == "" {
-		client, err := kubeconfigutil.ClientSetFromFile(kubeconfigPath)
+	if cfgPath == "" && client != nil {
+		internalcfg, err := configutil.FetchInitConfigurationFromCluster(client, printer, logPrefix, false, false)
```

Comment:
@neolit123 if we give the printer to `FetchInitConfigurationFromCluster`, `-o json` will not print the message then.

Reply:
if something works, great. let's apply the fix.

Reply:
I split the previous commit into two commits.
- the first commit fixes the JSON output (`-o json` will not print the message)
- the second commit adds some tests.

Reply:
> @neolit123 if we give the printer to FetchInitConfigurationFromCluster, -o json will not print the message then.

`+`

> I split the previous commit into two commits.
>
> the first commit fixes the JSON output (-o json will not print the message)
> the second commit adds some tests.


this seems OK, but can we also fix `upgrade plan -o json`?